src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -1214,15 +1214,14 @@ internal bool TryPeekByte(out byte value)
 
         // Takes a byte array, an offset, and a len and fills the array from the offset to len number of
         // bytes from the in buffer.
-        public bool TryReadByteArray(byte[] buff, int offset, int len)
+        public bool TryReadByteArray(Span<byte> buff, int len)
         {
-            int ignored;
-            return TryReadByteArray(buff, offset, len, out ignored);
+            return TryReadByteArray(buff, len, out _);
         }
 
         // NOTE: This method must be retriable WITHOUT replaying a snapshot
         // Every time you call this method increment the offset and decrease len by the value of totalRead
-        public bool TryReadByteArray(byte[] buff, int offset, int len, out int totalRead)
+        public bool TryReadByteArray(Span<byte> buff, int len, out int totalRead)
```

Comment:
Can we get rid of the length parameter too?

Reply:
No, there are cases where an empty buffer and non-zero length are passed in to skip a number of items, The skip length isn't predetermined so it may not be safe to stackalloc.